internal/controller/progressive/analysis.go

```diff
@@ -134,5 +142,95 @@ func CreateAnalysisRun(ctx context.Context, analysis apiv1.Analysis, existingUpg
 		return err
 	}
 
+	numaLogger.WithValues("AnalysisRunName", analysisRun.Name).Debug("Successfully created AnalysisRun")
+
 	return nil
 }
+
+func PerformAnalysis(
+	ctx context.Context,
+	existingUpgradingChildDef *unstructured.Unstructured,
+	rolloutObject ProgressiveRolloutObject,
+	analysis apiv1.Analysis,
+	analysisStatus *apiv1.AnalysisStatus,
+	c client.Client,
+) (*apiv1.AnalysisStatus, error) {
+	if analysisStatus == nil {
+		return analysisStatus, errors.New("analysisStatus not set")
+	}
+
+	analysisRun := &argorolloutsv1.AnalysisRun{}
+
+	analysisRunName := fmt.Sprintf("%s-%s", strings.ToLower(existingUpgradingChildDef.GetKind()), existingUpgradingChildDef.GetName())
```

Comment:
I realized the theoretical case that somebody has an Asset which contains a Pipeline named "my-event-consumer" and a MonoVertex named "my-event-consumer". We don't want these to have the same AnalysisRun names or there will be issues. Therefore, I am prefixing them with the kind.